particle-php/Validator
# Feature: Result a validation result

### What

To keep the validation API result clean and make sure you have everything you want after validation in a neat package, we moved the validation result to... ValidationResult! This is now returned by the `validate()` function, and allows you to do the following functions:

* `->isValid() : bool`
* `->isNotValid() : bool`
* `->getMessages() : array` an array with the validation messages
* `->getValues() : array` an array with the validated fields

### How to test

1. See that `validate()` now returns a ValidatonResult
1. See that you can call the 4 functions stated above from the ValidatonResult
1. See that all tests pass with the new `ValidatonResult->isValid()` functionality
1. See that the documentation is updated, and ValidationResult is used everywhere.
1. See that the code coverage and quality is still 100%

### ToDo

* [x] Make sure the coverage and quality is still 100%
* [x] Add better usage example

### Note

Version `v2.0.0` should be released after this is merged, because existing usage of particle/validator ~1 will break once updated.
